airflow/jobs/base_job.py

```diff
@@ -104,6 +108,11 @@ def __init__(self, executor=None, heartrate=None, *args, **kwargs):
         self.max_tis_per_query = conf.getint('scheduler', 'max_tis_per_query')
         super().__init__(*args, **kwargs)
 
+    @cached_property
+    def executor(self):
+        """Provided Executor which defaults to the one mentioned via Airflow configuration"""
+        return self._executor or ExecutorLoader.get_default_executor()
+
```

Comment:
```suggestion
    @cached_property
    def executor(self):
        return ExecutorLoader.get_default_executor()

    @property
    def executor_class(self):
        return self.executor.__class__.__name__
```

Reply:
Well this will cause the same issue, executor_class is stored in DB.

So when the Job model will be saved to DB it access self.executor_class which will execute self.execute and that will instantiate Executor, which we want to avoid

Reply:
Ah, I didn't realise that. In which case change this to a `@cached_property` too, and then set it in the `if executor:` block.

Reply:
Oh right no. Okay.

Reply:
Leave executor_class set as a normal attribute (as you had it) -- we just don't need `_executor` but can directly set `self.executor`